src/distance/JointAvoidanceConfigurationRanker.cpp

```diff
@@ -0,0 +1,60 @@
+#include "aikido/distance/JointAvoidanceConfigurationRanker.hpp"
+
+namespace aikido {
+namespace distance {
+
+using statespace::dart::ConstMetaSkeletonStateSpacePtr;
+using ::dart::dynamics::ConstMetaSkeletonPtr;
+
+//==============================================================================
+JointAvoidanceConfigurationRanker::JointAvoidanceConfigurationRanker(
+    ConstMetaSkeletonStateSpacePtr metaSkeletonStateSpace,
+    ConstMetaSkeletonPtr metaSkeleton,
+    const std::vector<statespace::StateSpace::State*> ikSolutions)
+  : ConfigurationRanker(metaSkeletonStateSpace, metaSkeleton, ikSolutions)
+{
+  // Do nothing
+}
+
+//==============================================================================
+double JointAvoidanceConfigurationRanker::evaluateIKSolution(
+    statespace::StateSpace::State* solution) const
+{
+  Eigen::VectorXd solutionPosition(mMetaSkeletonStateSpace->getDimension());
+  mMetaSkeletonStateSpace->convertStateToPositions(
+      mMetaSkeletonStateSpace->cloneState(solution), solutionPosition);
+
+  auto lowerLimitPosition = mMetaSkeleton->getPositionLowerLimits();
+  auto upperLimitPosition = mMetaSkeleton->getPositionLowerLimits();
+
+  for (std::size_t i = 0; i < mMetaSkeletonStateSpace->getDimension(); ++i)
+  {
+    if (lowerLimitPosition[i] == -dart::math::constantsd::inf()
+        || upperLimitPosition[i] == dart::math::constantsd::inf())
+    {
+      lowerLimitPosition[i] = solutionPosition[i];
+      upperLimitPosition[i] = solutionPosition[i];
+    }
+  }
+
+  auto lowerLimitState = mMetaSkeletonStateSpace->createState();
+  mMetaSkeletonStateSpace->convertPositionsToState(
+      lowerLimitPosition, lowerLimitState);
+
+  auto upperLimitState = mMetaSkeletonStateSpace->createState();
+  mMetaSkeletonStateSpace->convertPositionsToState(
+      lowerLimitPosition, upperLimitState);
+
+  auto dmetric = createDistanceMetricFor(
```

Comment:
Should this distance metric be a member variable in `ConfigurationRanker` that we can reuse?